src/test/java/io/jenkins/plugins/github/checks/api/ChecksDetailsTest.java

```diff
@@ -0,0 +1,90 @@
+package io.jenkins.plugins.github.checks.api;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import io.jenkins.plugins.github.checks.ChecksConclusion;
+import io.jenkins.plugins.github.checks.ChecksStatus;
+import io.jenkins.plugins.github.checks.api.ChecksDetails.ChecksDetailsBuilder;
+
+import static org.assertj.core.api.Assertions.*;
+
+/**
+ * Tests the class {@link ChecksDetails}.
+ */
+class ChecksDetailsTest {
+    private final String CHECK_NAME = "Jenkins";
+
+    @Test
+    void shouldCreateWhenBuildOnlyWithConstructor() {
+        ChecksDetails details = new ChecksDetailsBuilder(CHECK_NAME, ChecksStatus.QUEUED)
+                .build();
+
+        assertThat(details.getName()).isEqualTo(CHECK_NAME);
+        assertThat(details.getStatus()).isEqualTo(ChecksStatus.QUEUED);
+        assertThat(details.getDetailsURL().isPresent()).isFalse();
+        assertThat(details.getConclusion().isPresent()).isFalse();
+        assertThat(details.getOutputs().isPresent()).isFalse();
+        assertThat(details.getActions().isPresent()).isFalse();
+    }
+
+    @Test
+    void shouldCreateWhenBuildWithAllFields() {
+        // TODO: Changes may required here after refactoring Output and Action
+        List<Output> outputs = new ArrayList<>();
+        List<Action> actions = new ArrayList<>();
+        outputs.add(new Output());
+        outputs.add(new Output());
+        actions.add(new Action());
+
+        final String detailsURL = "ci.jenkins.io";
+        ChecksDetailsBuilder builder = new ChecksDetailsBuilder(CHECK_NAME, ChecksStatus.COMPLETED)
+                .withDetailsURL(detailsURL)
+                .withConclusion(ChecksConclusion.SUCCESS)
+                .withOutputs(outputs)
+                .withActions(actions);
+
+        ChecksDetails details = builder.build();
+        assertThat(details.getName()).isEqualTo(CHECK_NAME);
+        assertThat(details.getStatus()).isEqualTo(ChecksStatus.COMPLETED);
+        assertThat(details.getDetailsURL().isPresent()).isTrue();
+        assertThat(details.getDetailsURL().get()).isEqualTo(detailsURL);
+        assertThat(details.getOutputs().isPresent()).isTrue();
+        assertThat(details.getOutputs().get().size()).isEqualTo(2);
```

Comment:
AssertJ heavily supports the tell don't ask principle 😄 

```suggestion
        assertThat(details.getOutputs()).hasSize(2);
```